### PR TITLE
Update interact mode stack items headers to include card type icon

### PR DESCRIPTION
**Pull request by lukemelia**
- Use new styling for interact mode card header
- Moves RealmIcon component to boxel-ui
- Introduces CardHeader component

image
image
image

### CODE REVIEW

File: packages/host/app/lib/current-run.ts [hunk 1]
```diff
@@ -700,7 +700,8 @@ function assertURLEndsWithJSON(url: URL): URL {
 function unwrap(html: string): string {
   return html
     .trim()
-    .replace(/^<div ([^<]*\n)/, '')
+    .replace(/^<div ([^>]*>)/, '')
+    .trim()
```
Comment (lukemelia) [inline packages/host/app/lib/current-run.ts:704]: My change to CardContainer exposed what I concluded was a bug in this method.

**Review by burieberry — COMMENTED** [review]
A couple of things I observed on production preview:
- Header and contrasting font color in cards that have custom header color is no longer working
- The text on the ai-chat skills button on hover should be left-aligned and not centered
- The card type icon is too small, in the latest design max-width or max-height is 20px
- When the save message appears underneath, the text height doesn't seem to fit in the available space and the message gets cropped in the bottom

**Comment by lukemelia** [comment]
@burieberry thanks for catching those issues. How did you reproduce the issue with the saving message? It seems to workout OK for me: 
image

**Review by burieberry — APPROVED** [review]
Looks good

**Pull request merged**